Source/WebCore/platform/network/HTTPParsers.cpp

```diff
@@ -610,32 +610,49 @@ OptionSet<ClearSiteDataValue> parseClearSiteDataHeader(const ResourceResponse& r
     return result;
 }
 
-bool parseRange(StringView range, long long& rangeOffset, long long& rangeEnd, long long& rangeSuffixLength)
+// Implements <https://fetch.spec.whatwg.org/#simple-range-header-value>.
+bool parseRange(StringView range, bool allowWhitespace, long long& rangeStart, long long& rangeEnd)
 {
-    // The format of "Range" header is defined in RFC 2616 Section 14.35.1.
-    // http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35.1
-    // We don't support multiple range requests.
+    rangeStart = rangeEnd = -1;
 
-    rangeOffset = rangeEnd = rangeSuffixLength = -1;
+    int index;
+
+    // FIXME: this whole function could be more efficient by walking through the range value once.
+    if (!allowWhitespace) {
+        // Only 0x20 and 0x09 matter as newlines are already gone by the time we parse a header value.
+        index = range.find(' ');
```

Comment:
It is a bit confusing to use the function scoped index variable.
I would just write it: `if (range.find(' ') != -1 return false;`
There is a find which takes an invocable, you might be able to use it to remove the FIXME.